File: 0_c_basics/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_change(void);

int main(void)
{
    // Prompt user for change
    int c = get_change();

    // Calculate minimum number of coins
    int quarters = c / 25;
    int dimes = (c % 25) / 10;
    int nickels = ((c % 25) % 10) / 5;
    int pennies = ((c % 25) % 10) % 5;
    int total = quarters + dimes + nickels + pennies;

    // Print total coins
    printf("%i\n", total);
}

int get_change(void)
{
    int x;
    do
    {
        x = get_int("Change owed :  ");
    }
    while (x < 0);
    return x;
}

File: 5_Python/sentimental-cash/cash.py
def main():

    # Prompt user for change
    c = -1
    while c < 0:
        try:
            c=float(input("Change: "))
            c=c*100
            print(c)
        except ValueError:
            print("Enter Numeric Value")

    # Calculate minimum number of coins greedily
    coins = 0
    if c >= 25:
        coins += int(c/25)
        c=(c%25)
    if c >= 10:
        coins += int(c/10)
        c=(c%10)
    if c>=5:
        coins += int(c/5)
        c=(c%5)
    if c >= 1:
        coins += int(c)
        c= (c%1)

    coins = int(coins)
    print(coins)

main()
